app/grandchallenge/cases/tasks.py

```diff
@@ -520,3 +522,27 @@ def _check_post_processor_result(*, post_processor_result, image):
 
     if created_ids not in [{str(image.pk)}, set()]:
         raise RuntimeError("Created image IDs do not match")
+
+
+@acks_late_micro_short_task(retry_on=(LockNotAcquiredException, RetryStep))
+@transaction.atomic
+def import_dicom_to_healthimaging(*, dicom_imageset_upload_pk):
+    upload = lock_model_instance(
+        app_label="cases",
+        model_name="DICOMImageSetUpload",
+        pk=dicom_imageset_upload_pk,
+    )
+
+    # the status to check here will ultimately have to be something like DICOMImageSetUploadStatusChoices.DEIDENTIFIED
+    if not upload.status == DICOMImageSetUploadStatusChoices.INITIALIZED:
+        raise RuntimeError(
+            "Upload is not ready for importing into HealthImaging."
+        )
+
+    upload.status = DICOMImageSetUploadStatusChoices.STARTED
+    upload.save()
+
+    try:
+        on_commit(upload.start_dicom_import_job)
+    except RetryStep:
+        raise
```

Comment:
The try ... except block doesn't have any effect as nothing is happening in the except, so everything will just get raised. The `on_commit` is also unnecessary as you are not calling a celery task nor does the call need to be deferred (we are anyway at the end of the transaction here):

```suggestion
    upload.start_dicom_import_job()
```

Reply:
I somehow thought the RetryStep had to be raised again for the celery task to pick it up properly, but on second thought that does not make sense.